diff -urP ekg-20020719.orig/Makefile.in ekg-20020719/Makefile.in
--- ekg-20020719.orig/Makefile.in	Fri Jul 19 23:20:23 2002
+++ ekg-20020719/Makefile.in	Fri Jul 19 23:23:19 2002
@@ -33,18 +33,18 @@
 	cd src && $(MAKE) install
 
 install-ekg-themes:
-	$(INSTALL) -d $(datadir)/ekg/themes
-	$(INSTALL) -m 644 themes/*.theme $(datadir)/ekg/themes
+	$(INSTALL) -d $(DESTDIR)$(datadir)/ekg/themes
+	$(INSTALL) -m 644 themes/*.theme $(DESTDIR)$(datadir)/ekg/themes
 
 install-ekg-man:
-	$(INSTALL) -d $(mandir)/man1
-	$(INSTALL) -m 644 docs/ekg.man.en $(mandir)/man1/ekg.1
-	$(INSTALL) -d $(mandir)/pl/man1
-	$(INSTALL) -m 644 docs/ekg.man.pl $(mandir)/pl/man1/ekg.1
+	$(INSTALL) -d $(DESTDIR)$(mandir)/man1
+	$(INSTALL) -m 644 docs/ekg.man.en $(DESTDIR)$(mandir)/man1/ekg.1
+	$(INSTALL) -d $(DESTDIR)$(mandir)/pl/man1
+	$(INSTALL) -m 644 docs/ekg.man.pl $(DESTDIR)$(mandir)/pl/man1/ekg.1
 
 install-ekgsearch:
-	$(INSTALL) -d $(bindir)
-	$(INSTALL) -m 755 contrib/ekgsearch $(bindir)
+	$(INSTALL) -d $(DESTDIR)$(bindir)
+	$(INSTALL) -m 755 contrib/ekgsearch $(DESTDIR)$(bindir)
 
 #
 
