diff -Nur ekg-1.6/lib/Makefile.in ekg-1.6.LDFLAGS/lib/Makefile.in
--- ekg-1.6/lib/Makefile.in	2005-06-25 08:41:07.000000000 +0200
+++ ekg-1.6.LDFLAGS/lib/Makefile.in	2006-08-03 03:43:38.000000000 +0200
@@ -12,6 +12,7 @@
 
 CC = @CC@ 
 CFLAGS = @CFLAGS_LIBGADU@ -I. -I.. -Wall
+LDFLAGS = @LDFLAGS@
 LIBS = @LIBS_LIBGADU@
 
 MAKE = @MAKE@
@@ -56,7 +57,7 @@
 shared:	dep $(SHARED)
 
 $(SHARED): $(LOBJS)
-	$(CC) -shared -Wl,-soname,libgadu.so.$(VERSION_MAJOR) -o $(SHARED) $(LOBJS) $(LIBS) -lc
+	$(CC) $(LDFLAGS) -shared -Wl,-soname,libgadu.so.$(VERSION_MAJOR) -o $(SHARED) $(LOBJS) $(LIBS) -lc
 	$(LN_S) -f $(SHARED) $(SHAREDM)
 	$(LN_S) -f $(SHARED) $(SHAREDMM)
 
