--- configure.in	2007-03-11 13:27:04.000000000 +0100
+++ configure.in-new	2007-05-31 23:16:33.300670618 +0200
@@ -292,7 +292,7 @@
 
 if test "x$have_openssl" = "xyes"; then
 	OBJS="$OBJS simlite.o"
-	LDFLAGS="$OPENSSL_LIBS $LDFLAGS"
+	LIBS="$OPENSSL_LIBS $LIBS"
 	CFLAGS="$CFLAGS $OPENSSL_INCLUDES"
 	if test "x$enable_libgadu_openssl" != "xno"; then
 		LIBS_LIBGADU="$LIBS_LIBGADU $OPENSSL_LIBS"
@@ -394,7 +394,7 @@
 		AC_MSG_ERROR([You need ncurses and ncurses-devel packages.])
 	fi
 
-	LDFLAGS="$LDFLAGS $CURSES_LIBS"
+	LIBS="$LIBS $CURSES_LIBS"
 	CFLAGS="$CFLAGS $CURSES_INCLUDES"
 	echo "CURSES_INCLUDES=$CURSES_INCLUDES"
 fi
@@ -407,11 +407,11 @@
 		TERMCAP_LIB=""
 	else
 		AC_CHECK_LIB(termcap, tgetent, [TERMCAP_LIB="-ltermcap"])
-		LDFLAGS="$LDFLAGS $TERMCAP_LIB"
+		LIBS="$LIBS $TERMCAP_LIB"
 	fi
 		if test "x$TERMCAP_LIB" = "x" -a "x$CURSES_LIBS" = "x"; then
 		AC_CHECK_NCURSES
-		LDFLAGS="$LDFLAGS $CURSES_LIBS"
+		LIBS="$LIBS $CURSES_LIBS"
 		CFLAGS="$CFLAGS $CURSES_INCLUDES"
 	fi
      
@@ -421,7 +421,7 @@
 		AC_MSG_ERROR([You need both readline and readline-devel packages.])
 	fi
 
-	LDFLAGS="$READLINE_LIBS $LDFLAGS"
+	LIBS="$READLINE_LIBS $LIBS"
 	CFLAGS="$CFLAGS $READLINE_INCLUDES"
 
 	AC_CHECK_FUNCS([rl_set_prompt rl_filename_completion_function rl_get_screen_size rl_set_key rl_bind_key_in_map])
@@ -453,7 +453,7 @@
 		AC_CHECK_HEADERS(linux/soundcard.h,
 		[
 			AC_DEFINE(HAVE_VOIP, 1, [define if you want VOIP support])
-			LDFLAGS="-lgsm $LDFLAGS"
+			LIBS="-lgsm $LIBS"
 			OBJS="$OBJS voice.o"
 		])
 	fi
@@ -472,7 +472,7 @@
 		AC_CHECK_HEADERS(gif_lib.h,
 		[
 			AC_DEFINE(HAVE_LIBUNGIF, 1, [define if you have libungif])
-			LDFLAGS="-lungif $LDFLAGS"
+			LIBS="-lungif $LIBS"
 			have_libungif=yes
 		])
 	])
@@ -491,7 +491,7 @@
 		AC_CHECK_HEADERS(jpeglib.h,
 		[
 			AC_DEFINE(HAVE_LIBJPEG, 1, [define if you have libjpeg])
-			LDFLAGS="-ljpeg $LDFLAGS"
+			LIBS="-ljpeg $LIBS"
 			have_libjpeg=yes
 		])
 	])
@@ -505,7 +505,7 @@
   
 if test "x$have_python" = "xyes"; then
 	OBJS="$OBJS python.o"
-	LDFLAGS="$PYTHON_LIBS $LDFLAGS"
+	LIBS="$PYTHON_LIBS $LIBS"
 	CFLAGS="$CFLAGS $PYTHON_INCLUDES"
 fi
 
@@ -522,7 +522,7 @@
 		AC_CHECK_HEADERS(zlib.h,
 		[
 			AC_DEFINE(HAVE_ZLIB, 1, [define if you want compressed logs])
-			LDFLAGS="-lz $LDFLAGS"
+			LIBS="-lz $LIBS"
 			have_zlib=yes
 		])
 	])
@@ -566,7 +566,7 @@
  	    if test "x$have_aspell_libs" = "xyes"; then
 		if test "x$ui_ncurses_default" = "xyes"; then    
  		    AC_DEFINE(WITH_ASPELL, 1, [define if you want aspell support])
- 		    LDFLAGS="$LDFLAGS -laspell"		
+		    LIBS="$LIBS -laspell"
 		fi
 	    else 
 	        enable_aspell=no
--- lib/Makefile.in	2006-04-18 21:05:11.000000000 +0200
+++ lib/Makefile.in-new	2007-05-31 23:18:35.346799383 +0200
@@ -12,7 +12,8 @@
 
 CC = @CC@ 
 CFLAGS = @CFLAGS_LIBGADU@ -I. -I..
-LIBS = @LIBS_LIBGADU@
+LDFLAGS = @LDFLAGS@
+LIBS = @LIBS@ @LIBS_LIBGADU@
 
 MAKE = @MAKE@
 AR = @AR@
@@ -56,7 +57,7 @@
 shared:	dep $(SHARED)
 
 $(SHARED): $(LOBJS)
-	$(CC) -shared -Wl,-soname,libgadu.so.$(VERSION_MAJOR) -o $(SHARED) $(LOBJS) $(LIBS) -lc
+	$(CC) $(LDFLAGS) -shared -Wl,-soname,libgadu.so.$(VERSION_MAJOR) -o $(SHARED) $(LOBJS) $(LIBS) -lc
 	$(LN_S) -f $(SHARED) $(SHAREDM)
 	$(LN_S) -f $(SHARED) $(SHAREDMM)
 
--- src/Makefile.in	2007-04-28 22:02:22.000000000 +0200
+++ src/Makefile.in-new	2007-05-31 23:20:07.254747976 +0200
@@ -28,10 +28,10 @@
 
 ekg:	$(OBJS) comptime.c @libgadu_a@
 	$(CC) $(CFLAGS) -c -o comptime.o comptime.c
-	$(CC) $(CFLAGS) -o ekg $(OBJS) comptime.o -L../lib @lgadu@ @libgadu_a@ $(LDFLAGS) $(LIBS)
+	$(CC) $(LDFLAGS) $(CFLAGS) -o ekg $(OBJS) -L../lib @lgadu@ @libgadu_a@ $(LIBS)
 
 ioctld:	ioctld.c
-	$(CC) $(CFLAGS) ioctld.c -o ioctld $(LIBS) @IOCTLD_OBJS@
+	$(CC) $(LDFLAGS) $(CFLAGS) ioctld.c -o ioctld $(LIBS) @IOCTLD_OBJS@
 
 dep:	.depend
 
